modules/@apostrophecms/schema/lib/addFieldTypes.js

```diff
@@ -1158,6 +1153,22 @@ module.exports = (self) => {
           fail('withType property, ' + type + ', does not match the name of any piece or page type module.');
         }
       }
+
+      function validateSchema(_field) {
+        if (!_field.schema) {
+          return;
+        }
+        if (!Array.isArray(_field.schema)) {
+          fail('schema property should be an array if present at this stage');
+        }
+        self.validate(_field.schema, {
+          type: 'relationship',
+          subtype: _field.withType
+        });
+        if (!_field.fieldsStorage) {
+          _field.fieldsStorage = _field.name.replace(/^_/, '') + 'Fields';
+        }
+      }
```

Comment:
Added this function for readability. I just added a call to `self.validate`, fields inside relationships were not validated.